### PR TITLE
FileGlobs.py : Recursively extract .npmignore globs

### PR DESCRIPTION
**Pull request by prachi1210**
Looks for .npmignore in dirs and subdirs recursively and extract file globs

Closes https://github.com/coala/coala-quickstart/issues/109

### CODE REVIEW

**Comment by prachi1210** [comment]
@Udayan12167 can you please review this?

File: tests/generation/file_globs_npmignore_testfiles/other_folder/.npmignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#Start of npmignore
+*.html
+#End of npmignore
```
Comment (nkprince007) [inline tests/generation/file_globs_npmignore_testfiles/other_folder/.npmignore]: Newline at the end, please.

File: tests/generation/file_globs_npmignore_testfiles/sample_data/data/.npmignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+#Start of npmignore
+*.css
+#End of npmignore
```
Comment (nkprince007) [inline tests/generation/file_globs_npmignore_testfiles/sample_data/data/.npmignore]: Newline at the end, please.

File: tests/generation/FileGlobs.py [hunk 3]
```diff
@@ -97,6 +98,98 @@ def test_get_project_files_gitignore(self):
...
+        os.chdir("other_folder")
+        with open('.npmignore', "w") as file:
+            file.write("""
+#Start of npmignore
+*.html
+#End of npmignore""")
+        os.chdir(os.path.dirname(os.path.realpath(__file__)))
+        os.chdir("file_globs_npmignore_testfiles")
+        os.makedirs("sample_data", exist_ok=True)
+        os.chdir("sample_data")
```
Comment (nkprince007) [inline tests/generation/FileGlobs.py]: There seem to be too many `chdir` operations. Can't this be done with combined use of `os.makedirs` and `open` alone?

File: tests/generation/file_globs_npmignore_testfiles/.npmignore [hunk 4]
```diff
@@ -0,0 +1,10 @@
+
+# Start of npmignore
+build
+ignore.c
+/tests
+/upload.c
+/*.py
+*.pyc
+__pycache__
+# End of npmignore
```
Comment (nkprince007) [inline tests/generation/file_globs_npmignore_testfiles/.npmignore]: Newline at the end, please.

**Comment by hemangsk** [comment]
Marking as WIP, one conflict to resolve and build is failing which should get resolved when the tests are added.

**Comment by prachi1210** [comment]
Closing this PR because of too many conflicts. Will raise a new one with fresh changes

**Pull request closed without merging**